### ISSUE TITLE
[glib] build failure on macOS (gtk, gtkmm)

### ISSUE THREAD

**Issue by SamuelMarks** [microsoft/vcpkg#17216]
**Host Environment**
- OS: macOS 11.1 (20C69)
- Compiler: Apple clang version 12.0.0 (clang-1200.0.32.29)
Target: x86_64-apple-darwin20.2.0
Thread model: posix
InstalledDir: /Applications/Xcode.app/Contents/Developer/Toolchains/XcodeDefault.xctoolchain/usr/bin

**To Reproduce**
```
$ ./vcpkg install gtkmm
```

**Failure logs** 
```
Computing installation plan...
The following packages will be built and installed:
  * atk[core]:x64-osx -> 2.24.0-5#2
  * atkmm[core]:x64-osx -> 2.24.2#5
  * brotli[core]:x64-osx -> 1.0.9#1
  * bzip2[core]:x64-osx -> 1.0.8#1
  * cairo[core,freetype,gobject]:x64-osx -> 1.16.0#9
  * cairomm[core]:x64-osx -> 1.15.3#5
  * dirent[core]:x64-osx -> 1.23.2
  * expat[core]:x64-osx -> 2020-08-18
  * fontconfig[core]:x64-osx -> 2.13.1#4
  * freetype[brotli,bzip2,core,png,zlib]:x64-osx -> 2.10.4
  * gdk-pixbuf[core]:x64-osx -> 2.36.9-5#7
  * gettext[core]:x64-osx -> 0.21
  * glib[core]:x64-osx -> 2.52.3#25
  * glibmm[core]:x64-osx -> 2.52.1#14
  * gtk[core]:x64-osx -> 3.22.19-4
    gtkmm[core]:x64-osx -> 3.22.2#5
  * harfbuzz[core]:x64-osx -> 2.7.4#1
  * json-c[core]:x64-osx -> 2019-09-10#1
  * libepoxy[core]:x64-osx -> 1.5.5
  * libffi[core]:x64-osx -> 3.3#7
  * libiconv[core]:x64-osx -> 1.16#8
  * libsigcpp[core]:x64-osx -> 2.10-3#2
  * pango[core]:x64-osx -> 1.40.11#9
  * pangomm[core]:x64-osx -> 2.40.1#4
  * pcre[core]:x64-osx -> 8.44#8
  * pixman[core]:x64-osx -> 0.38.4#2
  * pthread[core]:x64-osx -> 3.0.0
  * pthreads[core]:x64-osx -> 3.0.0-6
  * tool-meson[core]:x64-osx -> 0.56.02
Additional packages (*) will be modified to complete this operation.
Detecting compiler hash for triplet x64-osx...
Could not locate cached archive: .cache/vcpkg/archives/d3/d3bb2ea9c79ce52dfbbe519a8548338af0c861a5.zip
Could not locate cached archive: .cache/vcpkg/archives/89/89ca6211bc1059776287b4e4be3ce2d85faeb2de.zip
Could not locate cached archive: .cache/vcpkg/archives/bd/bda237913dff5b2b5d7b5f7ddcd532802528dc94.zip
Could not locate cached archive: .cache/vcpkg/archives/fd/fde93e9abe83238355f6c2f1ed5878e772534f6d.zip
Could not locate cached archive: .cache/vcpkg/archives/ce/cefa8a2531bd9710744d2041a010c35438ca0e7a.zip
Could not locate cached archive: .cache/vcpkg/archives/51/51d74d952255b9a402a8fd5b2d3d31d9b7c0c9a7.zip
Could not locate cached archive: .cache/vcpkg/archives/64/64ecda2543b157dad01ebf2ed0094efbd1aea4ac.zip
Could not locate cached archive: .cache/vcpkg/archives/23/2319f60d75f32ae4cf7ff18d491efe227abb7adf.zip
Could not locate cached archive: .cache/vcpkg/archives/e7/e7411970a41dbff29c65cb21db3a26d5a2f7fd75.zip
Could not locate cached archive: .cache/vcpkg/archives/7f/7f5a587df2d8bd523336a175cf3626138659d5a2.zip
Could not locate cached archive: .cache/vcpkg/archives/95/95301ddd2a75f1610d1c28c975d256509a5f24f5.zip
Could not locate cached archive: .cache/vcpkg/archives/47/47d736c23dc4241de8118cdf9458a577fdfacff9.zip
Could not locate cached archive: .cache/vcpkg/archives/8c/8c4c8267d61cba2426605225e145dea292b3bc12.zip
Could not locate cached archive: .cache/vcpkg/archives/2b/2b08d16a395043a8aa98d4dc32c9770e7d365b8b.zip
Could not locate cached archive: .cache/vcpkg/archives/f3/f3e69f5140f40903c83448d44a46f9c33cfd0960.zip
Could not locate cached archive: .cache/vcpkg/archives/21/21d4b418d31266f50133f63887ebc68565109a96.zip
Could not locate cached archive: .cache/vcpkg/archives/44/4467ff9e39e54e728b1dac0fed31a1b0f33f294a.zip
Could not locate cached archive: .cache/vcpkg/archives/62/629e4c73723b923799ae8033b81a40bf0d171152.zip
Could not locate cached archive: .cache/vcpkg/archives/73/732c7f8b5ec8420ca19994548e64660b1f8a16db.zip
Could not locate cached archive: .cache/vcpkg/archives/a6/a6e15bb2e3a8f15c4061a5bb272c920551d96b6d.zip
Could not locate cached archive: .cache/vcpkg/archives/80/80c2c0eff2f4a47a21e6eed147fe89fc9ea4e78d.zip
Could not locate cached archive: .cache/vcpkg/archives/95/951852b4912c09b0ef9fdff1f1febd25480a2d28.zip
Could not locate cached archive: .cache/vcpkg/archives/18/186e78b63ee5406103f3dea70218395543db424c.zip
Could not locate cached archive: .cache/vcpkg/archives/95/952581e05067cc11564857b5d4036b1a0038072d.zip
Could not locate cached archive: .cache/vcpkg/archives/82/82bae75d304252acafee169f7ec00d8bc6b08f27.zip
Could not locate cached archive: .cache/vcpkg/archives/21/2155fdd8e6e2ff8a6e450f644755ff2d32493860.zip
Could not locate cached archive: .cache/vcpkg/archives/fe/fea97d66b96c628951c8b1f07f1eaf8cf8619b1f.zip
Could not locate cached archive: .cache/vcpkg/archives/90/90fe07c736d162ef50d12fa0feefa45d72604aa4.zip
Could not locate cached archive: .cache/vcpkg/archives/e0/e091064d1cfe5ddfebdf47673bf2ea0e38444988.zip
Starting package 1/29: libiconv:x64-osx
Building package libiconv[core]:x64-osx...
-- Performing post-build validation
-- Performing post-build validation done
Stored binary cache: .cache/vcpkg/archives/d3/d3bb2ea9c79ce52dfbbe519a8548338af0c861a5.zip
Building package libiconv[core]:x64-osx... done
Installing package libiconv[core]:x64-osx...
Installing package libiconv[core]:x64-osx... done
Elapsed time for package libiconv:x64-osx: 128.7 ms
Starting package 2/29: gettext:x64-osx
Building package gettext[core]:x64-osx...
-- Downloading https://ftp.gnu.org/pub/gnu/gettext/gettext-0.21.tar.gz -> gettext-0.21.tar.gz...
-- Extracting source ./vcpkg/downloads/gettext-0.21.tar.gz
-- Applying patch 0002-Fix-uwp-build.patch
-- Applying patch 0003-Fix-win-unicode-paths.patch
-- Using source at ./vcpkg/buildtrees/gettext/src/0.21-73e9069d8f.clean
-- Configuring x64-osx-dbg
-- Configuring x64-osx-rel
-- Configuring x64-osx-dbg
-- Configuring x64-osx-rel
-- Building x64-osx-dbg
-- Installing x64-osx-dbg
-- Building x64-osx-rel
-- Installing x64-osx-rel
-- Performing post-build validation
-- Performing post-build validation done
Stored binary cache: .cache/vcpkg/archives/89/89ca6211bc1059776287b4e4be3ce2d85faeb2de.zip
Building package gettext[core]:x64-osx... done
Installing package gettext[core]:x64-osx...
Installing package gettext[core]:x64-osx... done
Elapsed time for package gettext:x64-osx: 3.021 min
Starting package 3/29: libffi:x64-osx
Building package libffi[core]:x64-osx...
-- Downloading https://github.com/libffi/libffi/archive/v3.3.tar.gz -> libffi-libffi-v3.3.tar.gz...
-- Extracting source ./vcpkg/downloads/libffi-libffi-v3.3.tar.gz
-- Applying patch win64-disable-stackframe-check.patch
-- Applying patch win32-disable-stackframe-check.patch
-- Using source at ./vcpkg/buildtrees/libffi/src/v3.3-8557367958.clean
-- Configuring x64-osx-dbg
-- Configuring x64-osx-rel
-- Building x64-osx-dbg
-- Building x64-osx-rel
-- Fixing pkgconfig file: ./vcpkg/packages/libffi_x64-osx/lib/pkgconfig/libffi.pc
-- Fixing pkgconfig file: ./vcpkg/packages/libffi_x64-osx/debug/lib/pkgconfig/libffi.pc
-- Installing: ./vcpkg/packages/libffi_x64-osx/share/libffi/copyright
-- Performing post-build validation
-- Performing post-build validation done
Stored binary cache: .cache/vcpkg/archives/bd/bda237913dff5b2b5d7b5f7ddcd532802528dc94.zip
Building package libffi[core]:x64-osx... done
Installing package libffi[core]:x64-osx...
Installing package libffi[core]:x64-osx... done
Elapsed time for package libffi:x64-osx: 3.005 s
Starting package 4/29: pcre:x64-osx
Building package pcre[core]:x64-osx...
-- Downloading https://ftp.pcre.org/pub/pcre/pcre-8.44.zip -> pcre-8.44.zip...
-- Extracting source ./vcpkg/downloads/pcre-8.44.zip
-- Applying patch pcre-8.44_suppress_cmake_and_compiler_warnings-errors.patch
-- Applying patch export-cmake-targets.patch
-- Using source at ./vcpkg/buildtrees/pcre/src/pcre-8-5bdacbc1cb.clean
-- Configuring x64-osx-dbg
-- Configuring x64-osx-rel
-- Building x64-osx-dbg
-- Building x64-osx-rel
-- Fixing pkgconfig file: ./vcpkg/packages/pcre_x64-osx/lib/pkgconfig/libpcre.pc
-- Fixing pkgconfig file: ./vcpkg/packages/pcre_x64-osx/lib/pkgconfig/libpcre16.pc
-- Fixing pkgconfig file: ./vcpkg/packages/pcre_x64-osx/lib/pkgconfig/libpcre32.pc
-- Fixing pkgconfig file: ./vcpkg/packages/pcre_x64-osx/lib/pkgconfig/libpcrecpp.pc
-- Fixing pkgconfig file: ./vcpkg/packages/pcre_x64-osx/lib/pkgconfig/libpcreposix.pc
-- Fixing pkgconfig file: ./vcpkg/packages/pcre_x64-osx/debug/lib/pkgconfig/libpcre.pc
-- Fixing pkgconfig file: ./vcpkg/packages/pcre_x64-osx/debug/lib/pkgconfig/libpcre16.pc
-- Fixing pkgconfig file: ./vcpkg/packages/pcre_x64-osx/debug/lib/pkgconfig/libpcre32.pc
-- Fixing pkgconfig file: ./vcpkg/packages/pcre_x64-osx/debug/lib/pkgconfig/libpcrecpp.pc
-- Fixing pkgconfig file: ./vcpkg/packages/pcre_x64-osx/debug/lib/pkgconfig/libpcreposix.pc
-- Installing: ./vcpkg/packages/pcre_x64-osx/share/pcre/copyright
-- Performing post-build validation
-- Performing post-build validation done
Stored binary cache: .cache/vcpkg/archives/fd/fde93e9abe83238355f6c2f1ed5878e772534f6d.zip
Building package pcre[core]:x64-osx... done
Installing package pcre[core]:x64-osx...
Installing package pcre[core]:x64-osx... done
Elapsed time for package pcre:x64-osx: 35.62 s
Starting package 5/29: glib:x64-osx
Building package glib[core]:x64-osx...
-- Downloading https://ftp.gnome.org/pub/gnome/sources/glib/2.52/glib-2.52.3.tar.xz -> glib-2.52.3.tar.xz...
-- Extracting source ./vcpkg/downloads/glib-2.52.3.tar.xz
-- Applying patch use-libiconv-on-windows.patch
-- Applying patch arm64-defines.patch
-- Applying patch fix-arm-builds.patch
-- Applying patch fix_pkgconfig.patch
-- Using source at ./vcpkg/buildtrees/glib/src/2.52.3-9cb1890713.clean
CMake Deprecation Warning at scripts/cmake/vcpkg_check_features.cmake:182 (message):
  calling `vcpkg_check_features` without the `FEATURES` keyword has been
  deprecated.

      Please add the `FEATURES` keyword to the call.
Call Stack (most recent call first):
  ports/glib/portfile.cmake:36 (vcpkg_check_features)
  scripts/ports.cmake:142 (include)


-- Configuring x64-osx-dbg
CMake Error at scripts/cmake/vcpkg_execute_required_process.cmake:105 (message):
    Command failed: ./vcpkg/downloads/tools/cmake-3.19.2-osx/cmake-3.19.2-macos-universal/CMake.app/Contents/bin/cmake ./vcpkg/buildtrees/glib/src/2.52.3-9cb1890713.clean -DHAVE_SELINUX=OFF -DGLIB_VERSION=2.52.3 -DCMAKE_MAKE_PROGRAM=./vcpkg/downloads/tools/ninja-1.10.1-osx/ninja -DCMAKE_SYSTEM_NAME=Darwin -DBUILD_SHARED_LIBS=OFF -DVCPKG_CHAINLOAD_TOOLCHAIN_FILE=./vcpkg/scripts/toolchains/osx.cmake -DVCPKG_TARGET_TRIPLET=x64-osx -DVCPKG_SET_CHARSET_FLAG=ON -DVCPKG_PLATFORM_TOOLSET=external -DCMAKE_EXPORT_NO_PACKAGE_REGISTRY=ON -DCMAKE_FIND_PACKAGE_NO_PACKAGE_REGISTRY=ON -DCMAKE_FIND_PACKAGE_NO_SYSTEM_PACKAGE_REGISTRY=ON -DCMAKE_INSTALL_SYSTEM_RUNTIME_LIBS_SKIP=TRUE -DCMAKE_VERBOSE_MAKEFILE=ON -DVCPKG_APPLOCAL_DEPS=OFF -DCMAKE_TOOLCHAIN_FILE=./vcpkg/scripts/buildsystems/vcpkg.cmake -DCMAKE_ERROR_ON_ABSOLUTE_INSTALL_DESTINATION=ON -DVCPKG_CXX_FLAGS= -DVCPKG_CXX_FLAGS_RELEASE= -DVCPKG_CXX_FLAGS_DEBUG= -DVCPKG_C_FLAGS= -DVCPKG_C_FLAGS_RELEASE= -DVCPKG_C_FLAGS_DEBUG= -DVCPKG_CRT_LINKAGE=dynamic -DVCPKG_LINKER_FLAGS= -DVCPKG_LINKER_FLAGS_RELEASE= -DVCPKG_LINKER_FLAGS_DEBUG= -DVCPKG_TARGET_ARCHITECTURE=x64 -DCMAKE_INSTALL_LIBDIR:STRING=lib -DCMAKE_INSTALL_BINDIR:STRING=bin -D_VCPKG_ROOT_DIR=./vcpkg -D_VCPKG_INSTALLED_DIR=./vcpkg/installed -DVCPKG_MANIFEST_INSTALL=OFF -DCMAKE_OSX_ARCHITECTURES=x86_64 -DGLIB_SKIP_HEADERS=ON -DGLIB_SKIP_TOOLS=ON -G Ninja -DCMAKE_BUILD_TYPE=Debug -DCMAKE_INSTALL_PREFIX=./vcpkg/packages/glib_x64-osx/debug
    Working Directory: ./vcpkg/buildtrees/glib/x64-osx-dbg
    Error code: 1
    See logs for more information:
      ./vcpkg/buildtrees/glib/config-x64-osx-dbg-out.log
      ./vcpkg/buildtrees/glib/

config-x64-osx-dbg-err.log

Call Stack (most recent call first):
  scripts/cmake/vcpkg_configure_cmake.cmake:333 (vcpkg_execute_required_process)
  ports/glib/portfile.cmake:40 (vcpkg_configure_cmake)
  scripts/ports.cmake:142 (include)


Error: Building package glib:x64-osx failed with: BUILD_FAILED
Please ensure you're using the latest portfiles with `./vcpkg update`, then
submit an issue at https://github.com/Microsoft/vcpkg/issues including:
  Package: glib:x64-osx
  Vcpkg version: 2021-01-13-unknownhash

Additionally, attach any relevant sections from the log files above.
```

**Additional context**
[config-x64-osx-dbg-err.log](https://github.com/microsoft/vcpkg/files/6291716/config-x64-osx-dbg-err.log)
[config-x64-osx-dbg-out.log](https://github.com/microsoft/vcpkg/files/6291717/config-x64-osx-dbg-out.log)

**Comment by Neumann-A** [comment]
That is known:
https://github.com/microsoft/vcpkg/blob/fe112d8301e13fde9a25dc92d94a0c4a6dc3cf19/scripts/ci.baseline.txt#L413
Fixed by #13100

**Comment by NancyLi1013** [comment]
Hi @SamuelMarks 
Thanks for posting this issue.
Since it's duplicated to #9012, let's track this issue there. 

Thanks for your information and fix @Neumann-A.
